skia-macros/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+
+                            data
+                        }),
+                    Span::call_site(),
+                );
+
+                let native_has_name = Ident::new(
+                    &format!("C_{name}_has{native_name}"),
+                    Span::call_site(),
+                );
+
+                let get_name = Ident::new(&format!("get_{attr}"), Span::call_site());
+                let native_get_name = Ident::new(
+                    &format!("C_{name}_get{native_name}"),
+                    Span::call_site(),
+                );
+                let set_name = Ident::new(&format!("set_{attr}"), Span::call_site());
+                let native_set_name = Ident::new(
+                    &format!("C_{name}_set{native_name}"),
+                    Span::call_site(),
+                );
+
+                let getter = match [optional.is_some(), copy.is_some()] {
+                    [true, true] => quote! {
+                        pub fn #get_name(&self) -> Option<#ty> {
+                            unsafe {
+                                if sb::#native_has_name(self.native()) {
+                                    let #getter_name = sb::#native_get_name(self.native()).as_ref().map(|value| *value);
+
+                                    #getter_body
+                                } else {
+                                    None
+                                }
+                            }
+                        }
+                    },
+                    [true, false] => quote! {
+                        pub fn #get_name(&self) -> Option<&#ty> {
+                            unsafe {
+                                if sb::#native_has_name(self.native()) {
+                                    let #getter_name = sb::#native_get_name(self.native()).as_ref();
+
+                                    #getter_body
+                                } else {
+                                    None
+                                }
+                            }
+                        }
+                    },
+                    [false, true] => quote! {
+                        pub fn #get_name(&self) -> #ty {
+                            unsafe {
+                                let #getter_name = *sb::#native_get_name(self.native()).as_ref().unwrap_unchecked();
+
+                                #getter_body
+                            }
+                        }
+                    },
+                    [false, false] => quote! {
+                        pub fn #get_name(&self) -> &#ty {
+                            unsafe {
+                                let #getter_name = sb::#native_get_name(self.native()).as_ref().unwrap_unchecked();
+
+                                #getter_body
+                            }
+                        }
+                    }
+                };
+
+                quote! {
+                    #getter
+
+                    pub fn #set_name(&self, #setter_name: #ty) {
```

Comment:
For the sake of making things more rusty I feel like this would be nice, but in any case, LGTM!
```suggestion
                    pub fn #set_name(&mut self, #setter_name: #ty) {
```